src/ApplicationInsights.ServiceFabric.Native/Net45/IdUtilHelper.cs

```diff
@@ -0,0 +1,96 @@
+﻿namespace Microsoft.ApplicationInsights.ServiceFabric.Remoting.Activities
+{
+    using Microsoft.ServiceFabric.Services.Runtime;
+    using System;
+    using System.Reflection;
+
+    /// <summary>
+    /// A static helper class that bridges the calls via reflection to service fabric runtime's internal class
+    /// of IdUtil. That contains the essential logic for computing the method and interface ids.
+    /// </summary>
+    internal static class IdUtilHelper
+    {
+        private static Func<MethodInfo, int> computeIdFromMethodDelegate;
+        private static Func<Type, int> computeIdFromTypeDelegate;
+        private static Func<string, string, int> computeIdFromNamesDelegate;
+
+        static IdUtilHelper()
+        {
+            try
+            {
+                Type idUtilType = typeof(ServiceRuntime).Assembly.GetType("Microsoft.ServiceFabric.Services.Common.IdUtil");
+
+                // The ComputeId methods are all internal. However, we specify both NonPublic and Public, just in cases
+                // service fabric changed them to become public in the future, then hopefully this code would still work.
+                MethodInfo computeIdFromMethod = idUtilType?.GetMethod(
```

Comment:
are there any legal concerns for using private/internal reflection?

Reply:
Will check on that